src/main/java/org/opentripplanner/street/search/state/State.java

```diff
@@ -52,6 +52,13 @@ public class State implements AStarState<State, Edge, Vertex>, Cloneable {
   // we should DEFINITELY rename this variable and the associated methods.
   public double walkDistance;
 
+  // how far a sharing vehicle powered by battery has driven
+  public double traversedBatteryMeters;
+
+  //the available battery distance of a currently selected sharing vehicle
+  //setting a magic value of 0.0 to make sure
+  public double currentRangeMeters;
+
```

Comment:
As said in the developer meeting, I agree that the algorithm itself has to be evaluated firstmost.

With that said, you can save a considerable amount of memory by selecting different data types for the fields.
By decreasing the granularity with three bits (8 meters) you can represent values up to 2040 meters using a single byte. The values are still comparable, it's just that the unit is octa-meters.

The value has to be mangled (shifted and casted) to fit but those operations are basically free from a CPU perspective.

Reply:
In practice it is not so easy, but lets not go into discussion on HW and Java architecture before we know how to solve this. I have a strong feeling that we do not need to add anything to the state to solve this.